From 2c2b300edea3dea24d59dce0347a70d37994c135 Mon Sep 17 00:00:00 2001
From: "M. Vefa Bicakci" <vefa.bicakci@windriver.com>
Date: Wed, 10 Nov 2021 20:52:06 -0500
Subject: [PATCH] keepalived.spec: Disable dependency on snmp

This commit removes keepalived package's dependency on SNMP. This was
done for two reasons:

- keepalived's configure script fails with the following error message
  when attempting to check the sanity of the C compiler flags needed to
  link keepalived with the net-snmp library.

  configure:11981: checking whether C compiler supports flag "..." from Net-SNMP
  configure:11992: gcc -o conftest -O2 -g -pipe -Wall ...
  /usr/lib64/libnetsnmpmibs.so: undefined reference to `headerGetEntry'
  collect2: error: ld returned 1 exit status

  This occurs because of the mock build environment has a version of RPM
  (and hence librpm) that is more recent than the one shipped with
  CentOS 7.x, and librpm's "headerGetEntry" function has since been
  deprecated, but net-snmp library is expected to dynamically link
  against an RPM library with the "headerGetEntry" function defined.

  For the record, another colleague has encountered this issue in the
  past, but with lldp instead of keepalived:
    https://gist.github.com/ericho/049d1908f5d80485541e918515996702

- To resolve the aforementioned issue, one can build net-snmp library in
  StarlingX as well. net-snmp used to be included in StarlingX's integ
  repository, but it was removed with the following commit:
    https://review.opendev.org/c/starlingx/integ/+/765381

keepalived is being updated in StarlingX, because OpenStack's neutron
depends on keepalived, and the stock version in CentOS 7 is not
compatible with the recently updated iptables. A software architect
colleague confirmed that neutron does not need to use keepalived in
StarlingX's context. This change is made to avoid the need to re-package
a large number of dependencies due to the recent iptables update.

Signed-off-by: M. Vefa Bicakci <vefa.bicakci@windriver.com>
---
 SPECS/keepalived.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/keepalived.spec b/SPECS/keepalived.spec
index 13650da79d64..8087fd474585 100644
--- a/SPECS/keepalived.spec
+++ b/SPECS/keepalived.spec
@@ -1,4 +1,4 @@
-%bcond_without snmp
+%bcond_with snmp
 %bcond_without vrrp
 %bcond_without sha1
 %bcond_with iptables
-- 
2.29.2

